cli-helpers/src/commands/migrate-avl-subscriptions.ts

```diff
@@ -0,0 +1,142 @@
+import { Uint8ArrayBlobAdapter } from "@smithy/util-stream";
+import csvToJson from "convert-csv-to-json";
+import inquirer from "inquirer";
+import { STAGES, STAGE_OPTION, invokeLambda } from "../utils";
+
+interface BodsSubscription {
+    dataset_id: string;
+    organisation_id: string;
+    url_link: string;
+    username: string;
+    password: string;
+    status: string;
+}
+
+interface Subscription {
+    id: string;
+    publisherId: string;
+    url: string;
+    username: string;
+    password: string;
+}
+
+const generateLambdaPayload = (
+    producerEndpoint: string,
+    username: string,
+    password: string,
+    subscriptionId: string,
+    publisherId: string,
+) => {
+    return {
+        body: `{\"dataProducerEndpoint\": \"${producerEndpoint}\",\"description\": \"Subscription for ${username}\",\"shortDescription\": \"Subscription for ${producerEndpoint}\",\"username\": \"${username}\",\"password\": \"${password}\",\"subscriptionId\": \"${subscriptionId}\",\"publisherId\": \"${publisherId}\"}`,
+    };
+};
+
+export const migrateAvlSubscriptions = new Command("migrate-avl-subscriptions")
+    .addOption(STAGE_OPTION)
+    .option("--fileName <fileName>", "Subscriptions CSV file name")
+    .action(async (options) => {
+        let { stage, fileName } = options;
+
+        if (!stage) {
+            const responses = await inquirer.prompt<{ stage: string }>([
+                {
+                    name: "stage",
+                    message: "Select the stage",
+                    type: "list",
+                    choices: STAGES,
+                },
+            ]);
+
+            stage = responses.stage;
+        }
+
+        if (!fileName) {
+            const responses = await inquirer.prompt<{ fileName: string }>([
+                {
+                    name: "fileName",
+                    message: "Enter the file name for the subscriptions CSV file",
+                    type: "input",
+                },
+            ]);
+
+            fileName = responses.fileName;
+        }
+
+        const subscriptionsJson = csvToJson
+            .fieldDelimiter(",")
+            .getJsonFromCsv(`${fileName}.csv`) as unknown as BodsSubscription[];
+
+        const setOfSubscriptionUrlAndUsername = new Set();
+        const dataProducersToSubscribeTo: Subscription[] = [];
+
+        subscriptionsJson.map((subscription) => {
+            if (subscription.status === "live") {
+                setOfSubscriptionUrlAndUsername.add(`${subscription.url_link}, ${subscription.username}`);
+
+                dataProducersToSubscribeTo.push({
+                    id: subscription.dataset_id,
+                    publisherId: subscription.organisation_id,
+                    url: subscription.url_link,
+                    username: subscription.username,
+                    password: subscription.password,
+                });
+            }
+        });
+
+        subscriptionsJson.map((subscription) => {
+            if (
```

Comment:
I think it's worth adding a comment in here about _why_ we loop over the list again analysing the error status subscriptions, as it's not obvious from reading the code alone